server/app/models/grid_service.rb

```diff
@@ -244,12 +244,18 @@ def dependant_services
     grid = self.grid
     dependant = []
     dependant += grid.grid_services.where(:$or => [
-        {:volumes_from => {:$regex => /^#{self.name}-%s/}},
-        {:volumes_from => {:$regex => /^#{self.name}-\d+/}},
-        {:affinity => "service==#{self.name}"},
-        {:affinity => "service!=#{self.name}"},
-        {:net => {:$regex => /^container:#{self.name}-%s/}},
-        {:net => {:$regex => /^container:#{self.name}-\d+/}}
+        # if this is a stackless service, this also includes bare volumes_from service names in other stacks
+        {:volumes_from => {:$regex => /^#{self.name_with_stack}-(%s|\d+)/}},
```

Comment:
The `#{self.name_with_stack}` form is actually impossible to use for stack services because of how the [dependency filter](https://github.com/kontena/kontena/blob/3d608c58a0c0ce046e3a3f2065c39431b2fbc13f/server/app/services/scheduler/filter/dependency.rb#L33-L34) is written: `Cannot find applicable node for service instance development/volumes-from/backup-1: Did not find any nodes for service dependency volumes_from: ["volumes-from.redis-1"]`.

The way that this works for stackless services is also broken re #2982